### PR TITLE
Keep TaskSpec immutable even with run parameters

### PR DESCRIPTION
**Pull request by dimroc**
Fixes https://www.pivotaltracker.com/story/show/163667151

### CODE REVIEW

File: adapters/adapter.go [hunk 1]
```diff
@@ -119,7 +119,8 @@ func For(task models.TaskSpec, store *store.Store) (*PipelineAdapter, error) {
 		if err != nil {
 			return nil, fmt.Errorf("%s is not a supported adapter type", task.Type)
 		}
-		b := Bridge{BridgeType: bt, Params: &task.Params}
+		pcopy := task.Params
```
Comment (j16r) [inline adapters/adapter.go]: Are both of these copies needed? (TaskSpec and Params?)

**Comment by thodges-gh** [comment]
I receive the following error when trying to start the Chainlink node from this branch, which I do not experience when starting from the version-0.6 branch:

```
{"level":"fatal","ts":1549388027.505686,"caller":"logger/logger.go:177","msg":"Error creating root directory \"/chainlink\": mkdir /chainlink: permission denied","stacktrace":"github.com/smartcontractkit/chainlink/logger.Fatal\n\t/home/thomas/go/src/github.com/smartcontractkit/chainlink/logger/logger.go:177\ngithub.com/smartcontractkit/chainlink/logger.Fatalf\n\t/home/thomas/go/src/github.com/smartcontractkit/chainlink/logger/logger.go:187\ngithub.com/smartcontractkit/chainlink/store.newConfigWithViper\n\t/home/thomas/go/src/github.com/smartcontractkit/chainlink/store/config.go:98\ngithub.com/smartcontractkit/chainlink/store.NewConfig\n\t/home/thomas/go/src/github.com/smartcontractkit/chainlink/store/config.go:80\nmain.NewProductionClient\n\t/home/thomas/go/src/github.com/smartcontractkit/chainlink/main.go:29\nmain.main\n\t/home/thomas/go/src/github.com/smartcontractkit/chainlink/main.go:17\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:201"}
```

I do have permission to the target directory (keep in mind this works fine building version-0.6).

**Comment by dimroc** [comment]
Hey there @thodges-gh,
Considering the error below is related to file permissions and that these changes don't touch disk related activity, I think it might be something else. Shall we sync up offline?
```
Error creating root directory \"/chainlink\": mkdir /chainlink: permission denied
```

File: adapters/adapter.go (hunk 1 above)
Comment (dimroc) [reply adapters/adapter.go]: No, I've removed it.

**Pull request merged**